### DISCUSSION TITLE
Does hocuspocus support E2EE, or the possibility of implementing encryption?

### DISCUSSION THREAD

**Discussion by bxff** [ueberdosis/hocuspocus#520] [Q&A]
Would I need to implement my own custom Yjs backend, or could is it encryption implementable using extensions on hocuspocus or being planed anytime soon.

**Comment by janthurau** [comment]
hey @bxff, you can technically store anything in the ydoc, we don't need to know the real contents. You can just implement this on the provider side by passing only encrypted data to the Provider.

**Reply by raineorshine** [reply]
> you can technically store anything in the ydoc

Hi @janthurau. Can you explain what you mean here? Y.applyUpdate, Y.encodeStateAsUpdate, Y.mergeUpdates, etc, all require valid YJS updates, so I don't see how anything can be stored in a Y.Doc. Thanks.

**Reply by fuadsaud** [reply]
@raineorshine I believe the suggestion is to insert encrypted data into the shared types, as there's no "protocol-level" encryption support. So the Yjs updates are not encrypted, but the content they contain is.

**Reply by raineorshine** [reply]
I see. So basically `ymap.set(KEY, encrypt(VALUE))` for every value in every shared type?
